packages/view360/src/View360.ts

```diff
@@ -0,0 +1,1065 @@
+   *
+   * await viewer.init();
+   *
+   * console.log(viewer.initialized); // true
+   * ```
+   */
+  public get initialized() { return this._initialized; }
+  /**
+   * Instance of the Autoplay manager.
+   * You can also change {@link View360Options#autoplay} options with this.
+   * @ko Autoplay 기능의 매니저 인스턴스.
+   * 이 인스턴스를 통해 {@link View360Options#autoplay} 옵션을 변경하는 것도 가능합니다.
+   * @since 4.0.0
+   * @readonly
+   * @example
+   * ```ts
+   * // Disable autoplay
+   * viewer.autoplay.disable();
+   * ```
+   */
+  public get autoplay() { return this._autoplay; }
+  /**
+   * When this value is `true` and {@link View360Options#projection} is set, {@link View360#init init()} will be called automatically when instance is created.
+   * @ko 이 값이 `true`이고, {@link View360Options#projection}이 설정되었으면, 인스턴스 생성 시점에 자동으로 {@link View360#init init()}을 호출합니다.
+   * @default true
+   * @since 4.0.0
+   * @example
+   * ```ts
+   * import View360, { EquirectProjection, EVENTS } from "@egjs/view360";
+   *
+   * // viewer.init() is called on instance creation
+   * // But as `init` is asynchronous, you should wait for "ready" event if you want to do something after initialization.
+   * const viewer = new View360("#el_id", {
+   *   autoInit: true,
+   *   projection: new EquirectProjection({ src: "SRC_TO_URL" })
+   * });
+   *
+   * console.log(viewer.initialized); // false, as `init` is asynchronous
+   *
+   * viewer.once(EVENTS.READY, () => {
+   *   console.log(viewer.initialized); // true
+   * });
+   * ```
+   */
+  public get autoInit() { return this._autoInit; }
+  /**
+   * When `true`, {@link View360#resize} is called when the canvas size is changed.
+   * @ko `true`일 경우, 캔버스의 크기가 변경되었을 때 자동으로 {@link View360#resize}를 호출합니다.
+   * @default true
+   * @since 4.0.0
+   * @see View360#useResizeObserver
+   * @example
+   * ```ts
+   * const viewer = new View360("#el_id", {
+   *   autoResize: true
+   * });
+   *
+   * // This can trigger `viewer.resize()` if the canvas size was not 400px
+   * const canvas = viewer.renderer.canvas;
+   * canvas.style.width = "400px";
+   * ```
+   */
+  public get autoResize() { return this._autoResize; }
+  /**
+   * CSS selector for canvas element to render panorama image/video.
+   * The canvas element should be placed inside the root element. (Dont' have to be direct child)
+   * @ko 파노라마 이미지/비디오를 렌더링할 canvas 엘리먼트의 CSS 선택자
+   * 캔버스 엘리먼트는 루트 엘리먼트 내부에 있어야합니다. 루트 엘리먼트의 직계 자식 엘리먼트(Direct child element)일 필요는 없습니다.
+   * @default "canvas"
+   * @since 4.0.0
+   * @example
+   * ```html
+   * <div class="view360-container">
+   *   <canvas id="not_this_one"></canvas>
+   *   <!-- This will be selected -->
+   *   <canvas id="canvas_to_select"></canvas>
+   * </div>
+   * ```
+   *
+   * ```ts
+   * const viewer = new View360("#el_id", {
+   *   canvasSelector: "#canvas_to_select"
+   * });
+   * ```
+   */
+  public get canvasSelector() { return this._canvasSelector; }
```

Comment:
It would be better to be able to set both HTMLCanvasElement and selector(string) rather than one selector(string).

Reply:
That could be handy if we're planning to support the canvas element outside the container element.
That's not the case at the moment, so I'll pin this as a feature to be supported in our issue.